app/tests/integration/test_branched_addon_e2e.py

```diff
@@ -26,48 +57,22 @@ def test_branched_addon_e2e(
     fill_risks_page,
     signoff_and_ship,
     variables,
+    default_data,
 ):
     url = urlparse(selenium.current_url)
     experiment_url = f"{url.scheme}://{url.netloc}/api/v1{url.path}recipe/"
     experiment_json = requests.get(f"{experiment_url}", verify=False).json()
-    assert variables[experiment_type]["userFacingName"] in experiment_json["name"]
-    assert variables[experiment_type]["action_name"] == experiment_json["action_name"]
-    assert (
-        variables[experiment_type]["type"] == experiment_json["filter_object"][1]["type"]
-    )
+    api_json = APIDataClass(**experiment_json)
+    assert default_data.action_name == api_json.action_name
+    assert default_data.experiment_type == api_json.filter_object[1].type
+    assert default_data.channels.lower() in api_json.filter_object[1].channels
+    assert default_data.min_version == api_json.filter_object[2].versions[0]
+    assert default_data.max_version == api_json.filter_object[2].versions[-1]
+    assert default_data.user_facing_name in api_json.arguments.userFacingName
     assert (
-        variables[experiment_type]["channels"].lower()
-        == experiment_json["filter_object"][1]["channels"][0]
-    )
-    assert (
-        variables[experiment_type]["min_version"]
-        == f"{[item for item in experiment_json['filter_object'][2]['versions']][0]}.0"
-    )
-    assert (
-        variables[experiment_type]["max_version"]
-        == f"{[item for item in experiment_json['filter_object'][2]['versions']][-1]}.0"
-    )
-    assert (
-        variables[experiment_type]["userFacingName"]
-        in experiment_json["arguments"]["userFacingName"]
-    )
-    assert (
-        variables[experiment_type]["userFacingDescription"]
-        == experiment_json["arguments"]["userFacingDescription"]
-    )
-    assert len(variables[experiment_type]["branches"]) == len(
-        experiment_json["arguments"]["branches"]
+        default_data.user_facing_description in api_json.arguments.userFacingDescription
     )
-    for item in experiment_json["arguments"][
-        "branches"
-    ]:  # Loop over each item in the Branches secion
-        for num in range(
-            len(experiment_json["arguments"]["branches"])
-        ):  # Check each branch so we need to do the check for as many branches exis
-            try:
-                assert (
-                    item["slug"]
-                    == variables[experiment_type]["branches"][num]["branch_name"]
-                )
-            except AssertionError:
-                continue
+    assert len(default_data.branches) == len(api_json.arguments.branches)
+    api_branches = sorted(api_json.arguments.branches, key=lambda x: x.slug)
+    for count, data in enumerate(default_data.branches):
+        assert default_data.branches[count].branch_name == api_branches[count].slug
```

Comment:
This works but you could use the same pattern for the multipref where you pull it into a dictionary and then look it up by the branch slug.